content/blog/use-redfish-and-ipv6-to-find-lost-servers.md

```diff
@@ -0,0 +1,475 @@
+ilorest get Ipv4Addresses --filter Name="Manager Dedicated*" --json
+{
+  "IPv4Addresses": [
+    {
+      "Address": "192.168.1.45",
+      "AddressOrigin": "Static",
+      "Gateway": "192.168.1.1",
+      "SubnetMask": "255.255.252.0"
+    }
+  ]
+}
+
+ilorest logout
+```
+
+```shell
+# Use of PowerShell core (7) and Redfish basic authentication
+$Credentials = "username:password"
+$base64EncodedCreds = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($Credentials))
+$Headers = @{}
+$Headers["Content-Type"] = "application/json"
+$Headers["Authorization"] = "Basic $base64EncodedCreds"
+
+$Ipv6Address="FE80::9618:82FF:FE71:A07B%22"
+$Uri="https://[$Ipv6Address]/redfish/v1/Managers/1/EthernetInterfaces/1"
+
+$response = Invoke-WebRequest -Uri $Uri -Method GET -Headers $Headers -SkipCertificateCheck
+$JsonResponse = $response.Content | ConvertFrom-Json
+$JsonResponse.IPv4Addresses
+Address      AddressOrigin Gateway     SubnetMask
+-------      ------------- -------     ----------
+192.168.1.47 Static        192.168.1.1 255.255.252.0
+```
+
+### Multicast addresses
+
+An important concept in IPv6 is the absence of broadcast addresses. Instead, IPv6 uses multicast addresses. The exhaustive list of the pre-defined multicast addresses is present in <a href="https://datatracker.ietf.org/doc/html/rfc4291#section-2.7.1" target="_blank">RFC 4291</a>. Among them is the all nodes `FF02::1` address.
+
+From you brick-saver, when you issue a `ping FF02::1%Index` command, each and every IPv6 interfaces present on the LAN must reply with their link local address.
+By doing so, you can build the table of all your IPv6 neighbors. You will find more detail on how to use this table in the second scenario below.
+
+## Recovery process
+
+The recovery process presented in this blog post relies entirely on EUI-64 IPv6 link local addresses, and thus on MAC addresses. If you don't have
+the list of the affected iLO MAC addresses, you need to go through the next paragraph.
+
+In case you have that list and your systems are iLO 5 based or later with
+<a href="https://servermanagementportal.ext.hpe.com/docs/redfishservices/ilos/ilo6/ilo6_161/ilo6_network_resourcedefns161/#oemhpeipv6"
+target="_blank">RFC7217</a>
+disabled, you can skip the next paragraph and study the second scenario.
+
+### MAC address discovery
+
+The lost iLOs MAC address discovery is performed with the IPv6 neighbor table maintained in your brick-saver.
+
+The overall method is the following:
+
+1. Make sure affected servers are powered on. This step can only be done by a human being present in the computer room. Then hide their iLO port from the network by physically disconnecting the network cable. You could unplug the server power cables to achieve this goal, but this would hide also OS interfaces connected to the management LAN. Those interfaces would then re-appear in step 5 below, although they are not iLO interfaces and alter the discovered MAC address list.
```

Comment:
In the third sentence, put a comma after "Then" so it reads "Then, hide their..."
In the fourth sentence, modify it so it reads "... but this would **also hide** OS interfaces..." [placement of the word "also"]